README.md

```diff
@@ -1,6 +1,199 @@
 # plasmactl-bump
-Update the version of components that were updated in last commit
+Update the version of components that were updated in the last commit.
 
-### Excluded files
+### Excluded Files:
 - README.md
 - README.svg
+
+---
+
+### Work in Progress (WIP)
+
+**bump:** Updates the version of components that were modified after the last bump.  
+**Current bump logic:**
+1. Open the git repository.
+2. Check if the latest commit is not a bump commit.
+3. Collect a list of changed files until the previous bump commit is found (matching by author??). If the `last` option is passed, only take changes from the last commit. Prepare a map of resource objects to update.
+4. Get the short hash of the last commit.
+5. Iterate through the resource map and update the version of each.
+
+---
+
+### `bump --sync` or Propagation
+
+When resources are decoupled and stored in packages, there’s a need to preserve versions and maintain the current state of resources.
+`bump` works closely with `plasmactl compose`. The purpose of `compose` is to build the platform using resources from different places (domain repo, packages outside the repo).
+The purpose of `bump --sync` is to propagate the versions of a build to the current state.
+
+**How is it achieved?**
```

Comment:
Would be great to add this here, as intro:


- we look for artifact this way
- then we unarchive in in this dir
- then we compare it with current build to list resources with differences (any file)
- then for each updated/added/deleted resources we look for commit where changed happened in repos (both domain repo and packages repos) // this is also impacted by compose strategies
- then we compare each group\_vars and vaults of domain repo to look for updates/additions/deletions to retreive domain repo git commit where it happened
- then we order all these commits by date
- then for each we take commit hash and use it to successively propagate to resoures which depends on modified role + to resources which use updated group\_vars/vault var